.github/scripts/publish-rhel.sh

```diff
@@ -1,59 +1,80 @@
 #!/bin/bash
 
-get_containers()
+# We used to give project id in the queries but the new API asks for _ID field of the project.
+get_id_from_pid()
 {
-    PROJECT_ID=$1
-    VERSION=$2
-    RHEL_API_KEY=$3
-
-    RESPONSE=$( \
+    local PROJECT_ID=$1
+    local RHEL_API_KEY=$2
+    
+    local ID=$( \
         curl --silent \
-             --request POST \
-             --header "Content-Type: application/json" \
-             --header "Authorization: Bearer ${RHEL_API_KEY}" \
-             --data {} \
-             "https://connect.redhat.com/api/v2/projects/${PROJECT_ID}/tags?tags=${VERSION}")
-
-    echo "${RESPONSE}"
+             --request GET \
+             -H "X-API-KEY: ${RHEL_API_KEY}" \
+             "https://catalog.redhat.com/api/containers/v1/projects/certification/pid/${PROJECT_ID}" \
+             | jq -r '._id')
+    
+    echo "$ID"
 }
 
-get_container_build()
+get_image()
 {
-    PROJECT_ID=$1
-    VERSION=$2
-    RHEL_API_KEY=$3
-
-    BUILD=$(get_containers "${PROJECT_ID}" "${VERSION}" "${RHEL_API_KEY}" | jq -r '.tags[0]')
-    if [ "${BUILD}" == "null" ]; then
-        # TAG can be also stored as "${VERSION}, latest"
-        BUILD=$(get_containers "${PROJECT_ID}" "${VERSION}%2C%20latest" "${RHEL_API_KEY}" | jq -r '.tags[0]')
+    local PUBLISHED=$1
+    local ID=$2
+    local VERSION=$3
+    local RHEL_API_KEY=$4
+
+    if [ ${PUBLISHED} = "published" ]; then
```

Comment:
Everywhere in scrips better to use double [[ brackets instead of single [. It's more safer and doesn't required quotes inside. http://mywiki.wooledge.org/BashGuide/Practices#Bash_Tests

Reply:
I see, I will change it accordingly.